cookbooks/bcpc-hadoop/attributes/monitoring.rb

```diff
@@ -0,0 +1,109 @@
+# Refer Zabbix API(https://www.zabbix.com/documentation/2.2/manual/api/reference/) for the various attributes used
+# The following list is supposed to be a skeleton which should be extended/edited according to the deployment scenario
+# By default items, triggers actions are disabled and won't be monitored. To enable them, set "enable=true" in the json structure
+default["bcpc"]["hadoop"]["graphite"]["queries"] = {
+   'namenode' => {
+      'nnheapmem' => {
+         'type' => "jmx",
+         'query' => "memory.HeapMemoryUsage_committed",
+         'trigger_val' => "max(61,0)",
+         'value_type' => 3,
+         'trigger_cond' => "=0",
+         'trigger_name' => "NameNodeAvailability",
+         'enable' => false,
+         'trigger_desc' => "Namenode service seems to be down",
+         'severity' => 5,
+         'route_to' => "admin"
+      },
+      'numstaledn' => {
+         'type' => "jmx",
+         'query' => "nn_fs_name_system_state.FSNamesystemState.NumStaleDataNodes",
+         'enable' => false
+      }
+   },
+   'hbase_master' => {
+      'hbasenonheapmem' => {
+         'type' => "jmx",
+         'query' => "memory.NonHeapMemoryUsage_committed",
+         'trigger_val' => "max(61,0)",
+         'value_type' => 3,
+         'trigger_cond' => "=0",
+         'trigger_name' => "HBaseMasterAvailability",
+         'enable' => false,
+         'trigger_dep' => ["NameNodeAvailability"],
+         'trigger_desc' => "HBase master seems to be down",
+         'severity' => 5,
+         'route_to' => "admin"
+      },
+      'hbaseheapmem' => {
+         'type' => "jmx",
+         'query' => "memory.HeapMemoryUsage_committed",
+         'history_days' => 2,
+         'trend_days' => 30,
+         'enable' => false
+      },
+      'numrsservers' => {
+         'type' => "jmx",
+         'query' => "hbm_server.Master.numRegionServers",
+         'trigger_val' => "max(61,0)",
+         'value_type' => 3,
+         'trigger_cond' => "=0",
+         'trigger_name' => "HBaseRSAvailability",
+         'enable' => false,
+         'trigger_dep' => ["HBaseMasterAvailability"],
+         'trigger_desc' => "HBase region server seems to be down",
+         'severity' => 5,
+         'route_to' => "admin"
+      }
+   },
+   'bcpc-vm' => {
+      # Entry in suffix array added to hostname as suffix and 
+      # to key name, trigger name as suffix "_<hostname with suffix>"
+      # If specified, should be *first* entry for a host
+      'suffix' =>  [ "1", "2", "3" ], 
+      'memory_active' => {
+         'type' => "servers",
+         'query' => "memory.Active",
+         'trigger_val' => "max(61,0)",
+         'trigger_cond' => "=0",
+         'trigger_name' => "NodeAvailability",
+         'enable' => true,
+         'trigger_desc' => "Node seems to be down",
+         'severity' => 5,
+         'route_to' => "admin"
+      },
+      'disk_space_avail' => {
+         'type' => "servers",
+         'query' => "diskspace._disk_0.byte_avail",
```

Comment:
Can I do a glob pattern e.g. for disk_* a minimum is reaching 0 on this machine?